src/test/java/StringManipulationTest.java

```diff
@@ -55,95 +66,140 @@ public void testRemoveNthCharacter2() {
+        manipulatedstring.setString("Hello, World!");
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.removeNthCharacter(0, false);
+        });
     }
 
+
+
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello, World!");
+        String result = manipulatedstring.removeNthCharacter(3, false);
+        assertEquals("Helo Wrl!", result);
     }
 
+
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("world");
+        String result = manipulatedstring.removeNthCharacter(2, true);
+        assertEquals("w r d", result);
     }
 
+
     @Test
-    public void testGeSubStrings1() {
+    public void testGetSubStrings1() {
         manipulatedstring.setString("This is my string");
-        String [] sStings = manipulatedstring.getSubStrings(3, 4);
-
-        assertEquals(sStings[0], "my");
-        assertEquals(sStings[1], "string");
+        String[] substrings = manipulatedstring.getSubStrings(3, 4);
+        assertEquals("my", substrings[0]);
+        assertEquals("string", substrings[1]);
     }
 
     @Test
-    public void testGeSubStrings2() {
-        fail("Not yet implemented");
+    public void testGetSubStrings2() {
+        manipulatedstring.setString("This is a test string");
+        String[] result = manipulatedstring.getSubStrings(2, 4);
+        assertArrayEquals(new String[]{"is", "a", "test"}, result);
     }
+
+
     @Test
-    public void testGeSubStrings3() {
-        fail("Not yet implemented");
+    public void testGetSubStrings3() {
+        manipulatedstring.setString("Hello, World!");
+        String[] result = manipulatedstring.getSubStrings(1, 2);
+        assertArrayEquals(new String[]{"Hello,", "World!"}, result);
     }
+
+
     @Test
-    public void testGeSubStrings4() {
-        fail("Not yet implemented");
+    public void testGetSubStrings4() {
+        manipulatedstring.setString("mochi is the best dog");
+        String[] result = manipulatedstring.getSubStrings(2, 4);
+        assertArrayEquals(new String[]{"is", "the", "best"}, result);
     }
+
+
+
+
     @Test
-    public void testGeSubStrings5() {
-        fail("Not yet implemented");
+    public void testGetSubStrings5() {
+        manipulatedstring.setString("This is a test");
+        assertThrows(IndexOutOfBoundsException.class, () -> {
+            manipulatedstring.getSubStrings(4, 6);
+        });
     }
+
+
+
     @Test
-    public void testGeSubStrings6() {
-        fail("Not yet implemented");
+    public void testGetSubStrings6() {
+        manipulatedstring.setString("This is a test");
+        String[] result = manipulatedstring.getSubStrings(1, 3);
+        assertArrayEquals(new String[]{"This", "is", "a"}, result);
     }
 
+
     @Test
-    public void testRestoreString1()
-    {
+    public void testRestoreString1() {
```

Comment:
Need more unique test cases for testing `restoreString()` (e.g., `string.length !== array.length`, any of the indices in `int[] array` are negative or greater than or equal to the length of the string).